src/pages/Registration/Steps/getRegistrationState.ts

```diff
@@ -63,9 +63,13 @@ export function getRegistrationState(reg: SavedRegistration): {
 
   if (isDoneDocs(reg)) {
     const { ContactPerson: c, Registration: r } = reg;
+    const isSignedFSA =
+      r.Documentation.DocType === "FSA"
+        ? !!r.Documentation.SignedFiscalSponsorshipAgreement
+        : true;
     return {
       state: {
-        step: 4,
+        step: (isSignedFSA ? 4 : 3) as 4,
```

Comment:
only mark `4` is signed - allowing user to go to banking step.

Reply:
I understand that this works, but... casting `3 as 4` is just weird...
Looks like something that would be a part of an article titled "Bizarre code I've seen in my career" haha

Reply:
Casting `3 as 4` looks like something that would be make it into an article titled "Bizarre code I've seen in my career" haha
I understand that this works, but... it's just weird... Is there a way to make this work without such a hack?

Reply:
@ap-justin @misicnenad At the very very least, should we leave that all as is, def add have a comment explaining what's going on and why we are doing that. We'll thank ourselves 3 months later when we revisit it or a new team member starts working on it. :)

Reply:
> Casting 3 as 4 looks like something that would make it into an article titled "Bizarre code I've seen in my career" haha

`as 4` is type construct and doesn't actually make `3` transform to `4` (actual value) , it's needed to conform this particular block to 

```ts
type RegStep4 = {
  step: 4;
  data: Step4Data;
};
```
which accepts `documentation` , if I don't put this casting this step would of of type `3 | 4` which doesn't accept `documentation`. This would mean even documentation is previously done (but not signed), we can't put it in `getRegistrationState`

> . it's just weird... Is there a way to make this work without such a hack?

Of course I am thinking of a better way to handle this edge case, but considering this is on production - can't afford  to have it there for a long time, that's why I presented the least intrusive and fastest solution I know.  


> @ap-justin @misicnenad At the very very least, should we leave that all as is, def add have a comment explaining what's going on and why we are doing that. 

@SovereignAndrey, would add some temp comments.